spec/app/domain/authentication/authn-azure/validate_application_identity_spec.rb

```diff
@@ -0,0 +1,454 @@
+                                    .and_return([subscription_id_annotation, resource_group_annotation, mismatched_user_assigned_identity_annotation])
+          end
+          subject do
+            Authentication::AuthnAzure::ValidateApplicationIdentity.new(
+              resource_class:             resource_class,
+              validate_azure_annotations: validate_azure_annotations,
+            ).call(
+              account:               account,
+              service_id:            service_id,
+              username:              username,
+              xms_mirid_token_field: xms_mirid_token_field_system_identity,
+              oid_token_field:       oid_token_field
+            )
+          end
+
+          it "raises an error" do
+            expect { subject }.to raise_error(::Errors::Authentication::AuthnAzure::InvalidApplicationIdentity)
+          end
+        end
+
+        context "where the system assigned identity does not match the application identity" do
+          before(:each) do
+            allow(mocked_host).to receive(:annotations)
+                                    .and_return([subscription_id_annotation, resource_group_annotation, mismatched_system_assigned_identity_annotation])
+          end
+          subject do
+            Authentication::AuthnAzure::ValidateApplicationIdentity.new(
+              resource_class:             resource_class,
+              validate_azure_annotations: validate_azure_annotations,
+            ).call(
+              account:               account,
+              service_id:            service_id,
+              username:              username,
+              xms_mirid_token_field: xms_mirid_token_field_system_identity,
+              oid_token_field:       oid_token_field
+            )
+          end
+
+          it "raises an error" do
+            expect { subject }.to raise_error(::Errors::Authentication::AuthnAzure::InvalidApplicationIdentity)
+          end
+        end
+
+      end
+    end
+
+    context "An invalid application identity" do
+      context "that does not have required constraints present in annotations" do
+        before(:each) do
+          allow(mocked_host).to receive(:annotations)
+                                  .and_return([])
+        end
+        subject do
+          Authentication::AuthnAzure::ValidateApplicationIdentity.new(
+            resource_class:             resource_class,
+            validate_azure_annotations: validate_azure_annotations,
+          ).call(
+            account:               account,
+            service_id:            service_id,
+            username:              username,
+            xms_mirid_token_field: xms_mirid_token_field_system_identity,
+            oid_token_field:       oid_token_field
+          )
+        end
+        it "it raise an error" do
+          expect { subject }.to raise_error(::Errors::Authentication::AuthnAzure::MissingConstraint)
+        end
+      end
```

Comment:
let's make this test more specific and split it into 2 UTs:
1. missing subscription_id
2. missing resource_group

The current way is not string enough as you don't have both.

Reply:
also - the `it` clause should have the word `it` inside it.

Reply:
should or should not?

Reply:
should not